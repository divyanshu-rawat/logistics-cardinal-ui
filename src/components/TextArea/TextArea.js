import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { InputStyled } from '../Input';

// TODO: remove this line when styled-components is upgraded
// https://www.styled-components.com/docs/basics#extending-styles (as prop)
const TextAreaStyled = styled(InputStyled.withComponent('textarea'))`
  ${({ theme }) => `
    overflow: auto;
    resize: vertical;
    height: auto;
    line-height: ${theme.textAreaLineHeight};
  `};
`;

const TextArea = ({ hasError, placeholder, ...props }) => (
  <TextAreaStyled hasError={hasError} placeholder={placeholder} {...props} />
);

TextArea.propTypes = {
  /**
   * Determines if the textarea has error and applies the correct style to it
   */
  hasError: PropTypes.bool,
  /**
   * For instance, Some text...
   * */
  placeholder: PropTypes.string,
};

export { TextArea, TextAreaStyled };
