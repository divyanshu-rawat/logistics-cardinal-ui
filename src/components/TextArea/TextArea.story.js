import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { color, boolean, select, number } from '@storybook/addon-knobs';

import { FORM } from '../../../.storybook/sections';

// import withTests from '../../utils/withTests';
// import { rooster } from '../../themes';

import { TextArea } from '.';

storiesOf(`${FORM}TextArea`, module)
  // .addDecorator(withTests('Text/tests/index', 'TextStyled'))
  .add(
    'Simple',
    withInfo()(() => (
      <div style={{ width: '500px' }}>
        <TextArea placeholder="... ... ..." />
      </div>
    )),
  );
