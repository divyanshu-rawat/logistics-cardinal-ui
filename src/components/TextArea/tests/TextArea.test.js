import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { TextArea, TextAreaStyled } from '../TextArea';

describe('<TextArea />', () => {
  it('should render the <TextArea /> component', () => {
    const tree = toJson(shallow(<TextArea placeholder="" hasError={false} />));
    expect(tree).toMatchSnapshot();
  });

  it('should render the <TextAreaStyled />', () => {
    const tree = toJson(shallow(<TextAreaStyled theme={{}} />));
    expect(tree).toMatchSnapshot();
  });
});
