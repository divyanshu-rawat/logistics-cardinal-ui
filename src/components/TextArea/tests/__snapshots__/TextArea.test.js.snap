// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> should render the <TextArea /> component 1`] = `
<TextArea__TextAreaStyled
  hasError={false}
  placeholder=""
/>
`;

exports[`<TextArea /> should render the <TextAreaStyled /> 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "TextArea__TextAreaStyled-fyZRmq",
        "isStatic": false,
        "rules": Array [
          "display:block;width:100%;background-image:none;outline:0;height:",
          [Function],
          ";box-shadow:",
          [Function],
          ";font-size:",
          [Function],
          ";border-radius:",
          [Function],
          ";line-height:",
          [Function],
          ";font-weight:",
          [Function],
          ";padding:",
          [Function],
          ";color:",
          [Function],
          ";border:1px solid ",
          [Function],
          ";background-color:",
          [Function],
          ";",
          "transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;",
          ";&::placeholder{color:",
          [Function],
          ";}&:focus:not(:disabled),&:hover:not(:disabled){border-color:",
          [Function],
          ";}",
          [Function],
          ";",
        ],
      },
      "displayName": "TextArea__TextAreaStyled",
      "render": [Function],
      "styledComponentId": "TextArea__TextAreaStyled-fyZRmq",
      "target": "textarea",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={Object {}}
/>
`;
